### PR TITLE
feat: add course limit warning to admins

### PR DESCRIPTION
**Pull request by kiram15**
For ticket number [ENT-5005](https://openedx.atlassian.net/browse/ENT-5005), adding in a warning message for admins who are attempting to enroll in courses over the recommended course limit 
Screen Shot 2021-11-01 at 12 15 38 PM

# For all changes

- [x] Ensure adequate tests are in place (or reviewed existing tests cover changes)

# Only if submitting a visual change

- [x] Ensure to attach screenshots
- [x] Ensure to have UX team confirm screenshots

### CODE REVIEW

File: src/components/BulkEnrollmentPage/stepper/AddCoursesStep.jsx [hunk 1]
```diff
@@ -3,25 +3,28 @@ import PropTypes from 'prop-types';
...
 import { SearchData, SearchHeader } from '@edx/frontend-enterprise-catalog-search';
+import DismissibleCourseWarning from './DismissibleCourseWarning';
 
 import { configuration } from '../../../config';
 import { ADD_COURSES_TITLE, ADD_COURSE_DESCRIPTION } from './constants';
 
 import CourseSearchResults from '../CourseSearchResults';
 
 const currentEpoch = Math.round((new Date()).getTime() / 1000);
+const maxCourses = 7;
```
Comment (johnnagro) [inline src/components/BulkEnrollmentPage/stepper/AddCoursesStep.jsx:14]: can we make this all capitalized to make its usage super clear its a constant?

**Review by johnnagro — APPROVED** [review]
one nit improvement. otherwise looks great ⛵

File: src/components/BulkEnrollmentPage/stepper/AddCoursesStep.jsx (hunk 1 above)
Comment (binodpant) [reply src/components/BulkEnrollmentPage/stepper/AddCoursesStep.jsx:14]: +1

File: src/components/BulkEnrollmentPage/stepper/AddCoursesStep.jsx [hunk 2]
```diff
@@ -42,13 +45,18 @@ const AddCoursesStep = ({
   </>
 );
 
+AddCoursesStep.defaultProps = {
+  selectedCoursesNum: null,
```
Comment (binodpant) [inline src/components/BulkEnrollmentPage/stepper/AddCoursesStep.jsx:49]: think a 0 makes more sense as default?

**Comment by binodpant** [comment]
ah I see existing tests, if you like I can help with testing, but I figured something like this can work

```
const defaultProps = {
  subscriptionUUID: 'fakest-uuid',
  enterpriseSlug: 'sluggy',
  subscription: { uuid: 'foo', enterpriseCatalogUuid: 'bar' },
  enterpriseId: 'fancyEnt',
  selectedCoursesNum: 0 // -- add this to the defaultProps
};
in an existing test, verify that the dialog does not render
then in a separate test, add the prop and test for presence of the text, sth like
  it('displays that warning dialog text', () => {
    renderWithRouter(<StepperWrapper {...defaultProps, selectedCoursesNum=8} />);
    expect(screen.getByText('somethingskjdf;aksjdf;klasjdf')).toBeInTheDocument();
  });
```

**Pull request merged**